l2_proposer.py

```diff
@@ -24,6 +24,17 @@ def start(config: Config):
     log_file = open(log_file_path, "w")
     sys.stdout.flush()
 
+    # if remote deploy, check if clients are running on the same host and use localhost if true
+    if config.l1_node_remote_ip == config.l2_proposer_remote_ip:
+        l1_rpc = config.l1_rpc.replace(config.l1_node_remote_ip, "127.0.0.1")
+    else:
+        l1_rpc = config.l1_rpc
+
+    if config.l2_sequencer_remote_ip == config.l2_proposer_remote_ip:
+        l2_node_rpc = config.l2_node_rpc.replace(config.l2_sequencer_remote_ip, "127.0.0.1")
+    else:
+        l2_node_rpc = config.l2_node_rpc
+
```

Comment:
We'll probably want to implement this as a method on the config object.

Maybe
```
def ip_from(target_ip, from_ip)
     return "127.0.0.1" if target_ip == from_ip else target_ip    
```